packages/transaction-manager/src/transaction-manager.ts

```diff
@@ -236,31 +227,144 @@ export default class TransactionManager implements TransactionTypes.ITransaction
    */
   private async parseMultipleChannels(
     resultGetTx: DataAccessTypes.IReturnGetChannelsByTopic,
+    page?: number,
+    pageSize?: number,
+    updatedBetween?: TransactionTypes.ITimestampBoundaries,
   ): Promise<TransactionTypes.IReturnGetTransactionsByChannels> {
-    // Get the channels from the data-access layers to decrypt and clean them one by one
-    const result = await Object.keys(resultGetTx.result.transactions).reduce(
-      async (accumulatorPromise, channelId) => {
+    // Get all channel IDs and their latest timestamps
+    const channelsWithTimestamps = Object.entries(resultGetTx.result.transactions).map(
+      ([channelId, transactions]) => {
+        const filteredTransactions = transactions.filter((tx) => {
+          const timestamp = tx.timestamp || 0;
+          if (updatedBetween?.from && timestamp < updatedBetween.from) return false;
+          if (updatedBetween?.to && timestamp > updatedBetween.to) return false;
+          return true;
+        });
+        return {
+          channelId,
+          latestTimestamp: Math.max(...filteredTransactions.map((tx) => tx.timestamp || 0), 0),
+          hasValidTransactions: filteredTransactions.length > 0,
+          filteredTransactions,
+        };
+      },
+    );
+
+    // Only include channels that have transactions within the time boundaries
+    const allChannelIds = channelsWithTimestamps
+      .filter((channel) => channel.hasValidTransactions)
+      .sort((a, b) => b.latestTimestamp - a.latestTimestamp)
+      .map((channel) => channel.channelId);
+
+    const result = {
+      transactions: {} as Record<string, TransactionTypes.ITransaction[]>,
+      ignoredTransactions: {} as Record<string, TransactionTypes.ITransaction[]>,
+    };
+
+    const validChannels: string[] = [];
+
+    // Process all channels first
+    for (const channelId of allChannelIds) {
+      try {
+        // Find the channel with its filtered transactions
+        const channel = channelsWithTimestamps.find((c) => c.channelId === channelId);
+        if (!channel) continue;
+
         const cleaned = await this.channelParser.decryptAndCleanChannel(
           channelId,
-          resultGetTx.result.transactions[channelId],
+          channel.filteredTransactions,
         );
 
-        // await for the accumulator promise at the end to parallelize the calls to decryptAndCleanChannel()
-        const accumulator: any = await accumulatorPromise;
+        // When paginating, include channels based on validity
+        if (page !== undefined || pageSize !== undefined) {
```

Comment:
Will this fail if only `page` is defined but not `pageSize`, or vice versa?  Is there validation somewhere to ensure both are defined?

Reply:
I found the validation on line 301.

New question: Should this be `&&` instead of `||`?

Reply:
It should be AND. Fixing that